libavfilter/dnn_filter_common.c

```diff
@@ -17,6 +17,36 @@
  */
 
 #include "dnn_filter_common.h"
+#include "libavutil/avstring.h"
+
+static char **separate_expression(DnnContext *ctx, const char *expr, const char *val_sep, int *separated_nb)
+{
+    char *val, **parsed_vals = NULL;
+    int val_num = 0;
+    if (!expr || !val_sep || !separated_nb) {
+        return NULL;
+    }
+
+    if (!parsed_vals)
+        parsed_vals = av_mallocz(sizeof(*parsed_vals));
```

Comment:
do not check return value, and is av_malloc enough?

Reply:
OK, you mean i should put the error message in this func before return, right?
And av_malloc is enough, since i will realloc in line 38 if more mem needed.

Reply:
may use av_malloc_array since it is considered as an array, see below for av_realloc_f

Reply:
parsed_vals = av_...
if (!parsed_vals) {
    return NULL;
}

Reply:
OK, i forgot it